### ISSUE TITLE
Accessibility Issue in TinyMCE editor: Nested interactive controls are not announced by screen readers

### ISSUE THREAD

**Issue by kaushikpatidar** [tinymce/tinymce#7134]
We came across accessibility issue when we used the AXE plugin for testing the TinyMCE editor **v4** and **v5** as well.
**URLs**: 
1. [Issue in v4](https://www.tiny.cloud/docs-4x/demo/full-featured/)
2. [Issue in v5](https://www.tiny.cloud/docs/demo/full-featured/)

**Issue Description**: Nested interactive controls are not announced by screen readers. Ensure interactive controls are not nested
**More Details**: Focusable elements with an interactive control ancestor (any element that accepts user input such as button or anchor elements) are not announced by screen readers and create an empty tab stop. That is, you could tab to the element but the screen reader will not announce its name, role, or state. [More details here](https://dequeuniversity.com/rules/axe/4.3/nested-interactive?application=AxeChrome)

**Steps**:
1. Open the example/demo pages in browser. https://www.tiny.cloud/docs-4x/demo/full-featured/ or https://www.tiny.cloud/docs/demo/full-featured/ 
2. Scan the page for accessibility issues using the AXE tool plugin: https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd
3. Observe the errors.
I have attached screenshots for both the versions as well.
<img width="1680" alt="TinyMCE v4 Accessibility Nested Interactive Controls Issue" src="https://user-images.githubusercontent.com/43330166/130017281-a4edc577-e890-4264-b942-bfc73b90f8a4.png">
<img width="1680" alt="TinyMCE v5 Accessibility Nested Interactive Controls Issue" src="https://user-images.githubusercontent.com/43330166/130017289-c9a90fcf-145b-49e9-8880-8f32b4e050b8.png">
